unstructured/partition/doc.py

```diff
@@ -63,52 +63,74 @@ def partition_doc(
         This information will be reflected in elements' metadata and can be be especially
         useful when partitioning a document that is part of a larger document.
     """
-    # Verify that only one of the arguments was provided
-    if filename is None:
-        filename = ""
     exactly_one(filename=filename, file=file)
 
-    if len(filename) > 0:
-        _, filename_no_path = os.path.split(os.path.abspath(filename))
-        base_filename, _ = os.path.splitext(filename_no_path)
-        if not os.path.exists(filename):
-            raise ValueError(f"The file {filename} does not exist.")
+    last_modified = get_last_modified(filename, file, date_from_file_object)
```

Comment:
Extract the `last_modified` computation to its own function. This same logic appears in every partitioner and implementing it separately in each partitioner invites bugs arising from inconsistency. The extracted function can be moved to `partition.common` and used by other partitioners in a separate PR.